projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6081,13 +6081,42 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
                 const columnWidthCombined = parseInt(this._columnWidth, 10) * (column.colEnd ? column.colEnd - column.colStart : 1);
                 column.defaultWidth = columnWidthCombined + 'px';
             } else {
-                column.defaultWidth = this._columnWidth;
+                // D.K. March 29th, 2021 #9145 Consider min/max width when setting defaultWidth property
+                column.defaultWidth = this.getExtremumBasedColWidth(column);
                 column.resetCaches();
             }
         });
         this.resetCachedWidths();
     }
 
+    /**
+     * @hidden
+     * @internal
+     */
+    protected getExtremumBasedColWidth(column: IgxColumnComponent): string {
+        let width = this._columnWidth;
+        const minWidth = this._columnWidth.indexOf('%') === -1 ? column.minWidthPx : column.minWidthPercent;
```

Comment:
According to the docs, one could set the `columnWidth` property of the grid in the following fashion:
`<igx-grid ... [columnWidth]="100"></igx-grid>`
That setting will break the `getExtremumBasedColWidth` function, because this check `this._columnWidth.indexOf('%')` implicates that the _columnWidth property is a string. 
It should rather take into account scenarios when the variable has a numeric value as well.

Reply:
@3phase shouldn't throw an error now